### ISSUE TITLE
different output from keras model and the same model imported to dl4j

### ISSUE THREAD

**Issue by zhazhadi** [eclipse/deeplearning4j#5935]
I found that the predict output from the same model which is different in keras and dl4j.
I made code in keras like this:
`r = resnet50.ResNet50(weights='imagenet')`           
`r.summary()`         
`arr = np.ones([1, 224, 224, 3])` 
`print(r.predict(arr))`      
`r.save("/Users/r50.h5") `     

and the code in dl4j like this :
`val network = KerasModelImport.importKerasModelAndWeights("/Usersr/r50.h5", false)`
`println(network.summary())`
` var ll= Nd4j.ones(1,3,224,224)`
`val a = network.output(false, ll)`
`println(a)`

the output is different:
keras:    
[[1.93100277e-04 3.96246789e-04 5.83248002e-05 7.68540922e-05
  3.63481449e-05 7.13677728e-04 9.90606077e-06 4.73918699e-05
  2.56535768e-05 1.82255098e-04 9.14856151e-04 1.40060845e-04
  7.89922487e-05 9.31939649e-05 2.07006960e-05 7.86382661e-05
...]]

dl4j:    
0.0002,    0.0004, 3.7204e-5, 5.0386e-5, 2.6926e-5,    0.0006, 5.9236e-6, 3.2395e-5, 1.8612e-5,    0.0001,    0.0009,    0.0001, 5.2217e-5, 6.6117e-5, 1.2951e-5, 5.6148e-5, 8.5207e-5, 2.5419e-5,  5.509e-5,  4.523e-5,    0.0002,    0.0012,    0.0008,    0.0002, ...

**Comment by AlexDBlack** [comment]
@zhazhadi What version of DL4J is this?
If you are not on snapshots, upgrade to those and try again. There has been a number of fixes since the last release.
https://deeplearning4j.org/snapshots

**Comment by zhazhadi** [comment]
@AlexDBlack  I  have tried the version from 0.9.1 to 1.0.0-beta, and the problem is still exist.

**Comment by AlexDBlack** [comment]
As I said:
> If you are not on snapshots, upgrade to those and try again

1.0.0-beta is not snapshots. Snapshots are the latest build, with additional fixes that 1.0.0-beta doesn't have. See the page for details:
https://deeplearning4j.org/snapshots

**Comment by zhazhadi** [comment]
@AlexDBlack  now I'm on snapshots, but it can't help. The outputs from dl4j have no change.

**Comment by AlexDBlack** [comment]
Hm... we already have tests for validating resnet50 import... https://github.com/deeplearning4j/deeplearning4j/blob/451dd76b50355358dc176f2b704e98c43423c5b8/deeplearning4j/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/e2e/KerasModelEndToEndTest.java#L382

cc @maxpumperla

**Comment by maxpumperla** [comment]
@zhazhadi I'll look into this.

**Comment by maxpumperla** [comment]
might be related https://github.com/deeplearning4j/deeplearning4j/issues/5836#issuecomment-407888288

**Comment by maxpumperla** [comment]
hmm, I didn't look very closely at your output at first. this is a complex, 50 layer network with 2.5M parameters and it seems we have prediction differences in the 10e-4 range. Now, I realize that the output has 1000 classes in this case, but I'd like to investigate how model accuracy is affected by this. Might just come down to slight numerical differences in, say, the activation functions for DL4J and keras.

**Comment by agibsonccc** [comment]
Default data types are normally the issue for this. I've debugged this a few times now for people. If a future user has similar issues, please let us know in a new issue.
